src/OpenTelemetry/Metrics/MetricReader.cs

```diff
@@ -174,6 +182,152 @@ internal virtual void SetParentProvider(BaseProvider parentProvider)
             this.ParentProvider = parentProvider;
         }
 
+        internal Metric AddMetricWithNoViews(Instrument instrument)
+        {
+            var metricName = instrument.Name;
+            lock (this.instrumentCreationLock)
+            {
+                if (this.metricStreamNames.Contains(metricName))
+                {
+                    OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "Metric name conflicting with existing name.", "Either change the name of the instrument or change name using View.");
+                    return null;
+                }
+
+                var index = ++this.metricIndex;
+                if (index >= MaxMetrics)
+                {
+                    OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "Maximum allowed Metrics for the provider exceeded.", "Use views to drop unused instruments. Or configure Provider to allow higher limit.");
+                    return null;
+                }
+                else
+                {
+                    var metric = new Metric(instrument, this.preferredAggregationTemporality, metricName, instrument.Description);
+                    this.metrics[index] = metric;
+                    this.metricStreamNames.Add(metricName);
+                    return metric;
+                }
+            }
+        }
+
+        internal void RecordSingleStreamLongMeasurement(Metric metric, long value, ReadOnlySpan<KeyValuePair<string, object>> tags)
```

Comment:
These newly added methods seem to be a good fit for `MetricOwner`/`MetricWriter` 😆

Reply:
`MetricReader` does own the `Metric[]` now.

Reply:
I have split the `MetricReader` class into partial classes.